backend/app/services/chat/trigger/core.py

```
@@ -380,7 +385,12 @@ async def _stream_chat_response(
             logger.info(
                 f"[ai_trigger] Unified context processing completed: "
                 f"user_subtask_id={user_subtask_id}, "
-                f"extra_tools_count={len(extra_tools)}"
+                f"extra_tools_count={len(extra_tools)}, "
+                f"extra_tools={[t.name for t in extra_tools]}"
+            )
+        else:
+            logger.warning(
+                f"[ai_trigger] user_subtask_id is None, skipping context processing"
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)